### DISCUSSION TITLE
Writing to SD card, what happens when full?

### DISCUSSION THREAD

**Discussion by skilau** [gtxaspec/wz_mini_hacks#122] [General]
This is a question I have had for awhile, but figured I would cross the bridge when I get there.
But, what the heck.

With wz_mini_hacks sharing the SD card with the regular Wyze firmware, what happens when Wyze fills up the card with writing all the Video to the card?

Typically, Wyze will delete the old stuff (whole day?) to make room for the new stuff.

Does this mean there is a chance that any logging we might do, might not have space to do so. if we catch the card before Wyze has had a chance to purge the oldest days?

**Comment by capncybo** [comment]
From what I've seen... The more we use up via the hack, the less storage is actually available to the WYZE app itself.  Meaning the microSD-Card will appear smaller in size.  However, I'm not 100% sure & perhaps @gtxaspec will verify?

**Reply by gtxaspec** [reply]
the available storage on the memory card will be reduced by whatever size wz_mini takes up... currently under 100mb if I remember correctly.  

the total size of the memory card stays the same

**Reply by gtxaspec** [reply]
according to the camera documentation,

`"When the microSD card is full, it will overwrite the oldest footage and roll forward.`"

**Comment by gtxaspec** [comment]
@skilau logging will probably fail if the card is full.  I haven't tested this though

**Comment by virmaior** [comment]
speculating blindly here but I'm guessing that the same process that copies the [0-5][0-9].mp4 for the current minute over is the same one that checks for space remaining and deletes things.

If so that would mean the space for these collisions is 1 minute intervals and that the question of their possibility depends on the lower bound before wyze starts deleting stuff.

Across my five cameras, the free space is between 1.4 GB and 2.0GB on a 32gb SD card with continuous recording. Building rom this:

1. lower bound is around 1.4GB before it starts deleting things?
2. deletions happen to entire directories at once i.e. inside 20220621 it will delete 05 and 06 at the same time to free up space. And it keeps going until it hits some target value (maybe 2GB?)

My guess is that this means wyze will never fill up to the point where wz_mini doesn't have enough space.

**Comment by akivacp** [comment]
If we use the NFS option, it doesn't auto-delete. Is there a script we can run that will do the same job?
Should the script be on the server side or the camera side?
What would that script be? Sorry I'm not very good at scripts.
